com.unity.ml-agents/Runtime/Sensors/Reflection/ObservableAttribute.cs

```diff
@@ -185,7 +185,7 @@ static ISensor CreateReflectionSensor(object o, FieldInfo fieldInfo, PropertyInf
                 memberType = propertyInfo.PropertyType;
             }
 
-            if (!s_TypeToSensorInfo.ContainsKey(memberType))
+            if (!s_TypeToSensorInfo.ContainsKey(memberType) && !memberType.IsEnum)
```

Comment:
Another option would be to add enum types to s_TypeToSensorInfo as they're requested. That simplifies the logic a bit later.